src/poller.ts

```diff
@@ -0,0 +1,69 @@
+import type { Options } from './types'
+
+export default class Poller {
+  private timeoutId: number
+  private isRunning: boolean
+
+  constructor(
+    private fetchFlagsFn: () => Promise<any>,
+    private configurations: Options,
+    private pollInterval: number,
+    private maxAttempts: number
+  ) {}
+
+  public start(): void {
+    if (this.isPolling()) {
+      this.logDebug('Polling was requested but is already running - only one poller can run at a time.')
+      return
+    }
+
+    this.isRunning = true
+
+    // Don't start polling immediately as we have alreadu fetched flags on client initialization
+    window.setTimeout(() => this.poll(), this.configurations.pollingInterval)
```

Comment:
Strange - both the `esbuild` and Intellij TS language service can't transpile the code without `window` as it's assuming the `setTimeout` is the node implementation it appears which returns a `Timeout` instead of `number`

![Screenshot 2023-09-14 at 12 46 07](https://github.com/harness/ff-javascript-client-sdk/assets/23323369/73bf7442-fefc-4c67-8058-3c44cda0e31c)



![Screenshot 2023-09-14 at 12 44 09](https://github.com/harness/ff-javascript-client-sdk/assets/23323369/510901fd-3a5b-414b-93db-df3d86e5762b)

Reply:
Had a quick bit of research and ts.config seems to be set right for this - but may be missing something (as usual with TS/JS !)

![Screenshot 2023-09-14 at 12 52 31](https://github.com/harness/ff-javascript-client-sdk/assets/23323369/b4141002-4547-4151-b7a3-424007ff539c)

Reply:
It's just that you have `timeoutId` set as a `number`. Change it to `any` and you're sorted.

Reply:
Makes perfect sense.

Reply:
I'll remove the window refs and change the type